chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -2275,6 +2281,19 @@ impl PeerManagerActor {
         }
     }
 
+    // The routed message received its destination. If the timestamp of creation of this message is
+    // known, then update the corresponding latency metric histogram.
+    fn record_routed_msg_latency(&self, msg: &RoutedMessageV2) {
+        if let Some(created_at) = &msg.created_at {
+            if let Ok(created_at) = time::UtcSerializable::to_instant(created_at) {
+                let duration = self.clock.now_utc().sub(created_at);
```

Comment:
yes, that's the way I intended the clock to be used :+1: . Please also use self.clock when constructing the RoutedMessageV2.